packages/cojson/src/permissions.ts

```diff
@@ -285,38 +273,34 @@ function determineValidTransactionsForGroup(
       | MapOpPayload<`child_${CoID<RawGroup>}`, CoID<RawGroup>>;
 
     if (changes.length !== 1) {
-      logPermissionError("Group transaction must have exactly one change");
-      transaction.markInvalid();
+      transaction.markInvalid("Group transaction must have exactly one change");
       continue;
     }
 
     if (change.op !== "set") {
-      logPermissionError("Group transaction must set a role or readKey");
-      transaction.markInvalid();
+      transaction.markInvalid("Group transaction must set a role or readKey");
       continue;
     }
 
     if (change.key === "readKey") {
       if (!canAdmin(transactorRole)) {
-        logPermissionError("Only admins can set readKeys");
-        transaction.markInvalid();
+        transaction.markInvalid("Only admins can set readKeys");
         continue;
       }
 
       transaction.markValid();
       continue;
     } else if (change.key === "profile") {
       if (!canAdmin(transactorRole)) {
-        logPermissionError("Only admins can set profile");
-        transaction.markInvalid();
+        transaction.markInvalid("Only admins can set profile");
         continue;
       }
 
       transaction.markValid();
       continue;
     } else if (change.key === "root") {
       if (!canAdmin(transactorRole)) {
-        logPermissionError("Only admins can set root");
```

Comment:
Previously, it wasn't calling `tx.markInvalid`. If `tx.isValid` is false by default, calling `tx.markInvalid` now changes the `isValidated` to true. I think the current version is correct, but I'm asking for a double-check.

Reply:
Good catch!